completion/api/v1/views.py

```diff
@@ -148,3 +148,37 @@ def post(self, request, *args, **kwargs):  # pylint: disable=unused-argument
             }, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
 
         return Response({"detail": _("ok")}, status=status.HTTP_200_OK)
+
+
+class MilestonesExperimentView(APIView):
+    """
+    Handles API endpoints for the milestones experiments.
+    """
+
+    permission_classes = (permissions.IsAuthenticated, IsUserInUrl)
+    REQUIRED_KEYS = ['course_key', 'subsection_id']
+
+    def get(self, request, username):
+        """
+        Returns completion for a (user, subsection, course).
+
+        REST Endpoint format:
+        {
+            "course_key": course_key,
+            "subsection_id": subsection_id
+        }
+        The username should be passed in through the URL.
+        """
+        for key in self.REQUIRED_KEYS:
+            if key not in request.data:
+                raise ValidationError(_("Key '{key}' not found.".format(key=key)))
+
+        user = username
+        course_key = request.data['course_key']
```

Comment:
This is a GET request, which typically won't have a data payload in the request.  So you can either turn this into a post request, or change the format of the URL to something like `completion/<user>/<course>`